app_dart/lib/src/service/firestore.dart

```diff
@@ -27,4 +32,67 @@ class FirestoreService {
     );
     return FirestoreApi(client).projects.databases.documents;
   }
+
+  /// Writes [writes] to Firestore within a transaction.
+  ///
+  /// This is an atomic operation: either all writes succeed or all writes fail.
+  Future<CommitResponse> writeViaTransaction(List<Write> writes) async {
+    final ProjectsDatabasesDocumentsResource databasesDocumentsResource = await documentResource();
+    final BeginTransactionRequest beginTransactionRequest =
+        BeginTransactionRequest(options: TransactionOptions(readWrite: ReadWrite()));
+    final BeginTransactionResponse beginTransactionResponse =
+        await databasesDocumentsResource.beginTransaction(beginTransactionRequest, kDatabase);
+    final CommitRequest commitRequest =
+        CommitRequest(transaction: beginTransactionResponse.transaction, writes: writes);
+    return databasesDocumentsResource.commit(commitRequest, kDatabase);
+  }
+}
+
+/// Generates task documents based on targets.
+List<Document> targetsToTaskDocuments(Commit commit, List<Target> targets) {
+  final Iterable<Document> iterableDocuments = targets.map(
+    (Target target) => Document(
+      name: '$kDatabase/documents/tasks/${commit.sha}_${target.value.name}_1',
+      fields: <String, Value>{
+        'builderNumber': Value(integerValue: null),
```

Comment:
Are we sure this can be nullable? Is this the main identifier?

Reply:
Ah this might be updated at a later date.

Reply:
Exactly. We will update the entry later when status changes.